inc/spbc-settings.php

```diff
@@ -6428,3 +6441,66 @@ function spbc_settings_draw_save_button_bottom_block()
         __('Back to navigation', 'security-malware-firewall')
     );
 }
+
+/**
+ * Get HTML content for change role template
+ * @return void *echo
+ */
+function spbc_change_role_template()
+{
+    spbc_check_ajax_referer('spbc_secret_nonce', 'security');
+
+    $user_login = sanitize_text_field($_POST['data']['user_login']);
+    $role = sanitize_text_field($_POST['data']['role']);
+
+    $template = sprintf(
+        '<div> <p>%s <strong>%s</strong></p> <p>%s</p> %s <br> <button id="spbc_change_role_button">%s</button> </div>',
+        __('Change role for', 'security-malware-firewall'),
+        $user_login,
+        __('It will be applied immediately, be careful with this action.', 'security-malware-firewall'),
+        '%ROLE_SELECT%',
+        __('Change Role', 'security-malware-firewall')
+    );
+
+    $role_select = '<select name="role" id="spbc_role_select">';
+    $all_roles = wp_roles()->role_names;
+    foreach ($all_roles as $key => $value) {
+        if ($key == $role) {
+            $role_select .= '<option value="' . $key . '" selected>' . $value . '</option>';
+        } else {
+            $role_select .= '<option value="' . $key . '">' . $value . '</option>';
+        }
+    }
+    $role_select .= '</select>';
+
+    $template = str_replace('%ROLE_SELECT%', $role_select, $template);
+
+    echo $template;
+
+    die();
+}
+
+/**
+ * Change role for user
+ * @return void
+ */
+function spbc_change_role()
+{
+    spbc_check_ajax_referer('spbc_secret_nonce', 'security');
+
+    if ( ! current_user_can('promote_users') ) {
+        wp_send_json_error(array('message' => __('You are not allowed to change role for this user.', 'security-malware-firewall')));
+    }
+
+    $user_id = sanitize_text_field($_POST['user_id']);
+    $new_role = sanitize_text_field($_POST['new_role']);
+
+    $user = get_user_by('id', $user_id);
+    if ( ! $user ) {
+        wp_send_json_error(array('message' => __('User not found.', 'security-malware-firewall')));
+    }
+
+    $user->set_role($new_role);
```

Comment:
Is there any validation for $new_role? I do not think we should allow any string of role via POST, even if nonce provided

Reply:
Thanks, fixed https://github.com/CleanTalk/security-malware-firewall/pull/513/commits/d1a0e00780dc0fe7e3da80a396fb41ba66d97c08